### PR TITLE
update forge units package

### PR DESCRIPTION
**Pull request by pinzart90**
### Purpose

Update the ForgeUnits nuge package to 4.1.1

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/DynamoUnits/Quantity.cs [hunk 1]
```diff
@@ -12,9 +10,9 @@ namespace DynamoUnits
     /// </summary>
     public class Quantity
     {
-        internal readonly ForgeUnitsCLR.Quantity forgeQuantity;
```
Comment (mjkkirschner) [inline src/Libraries/DynamoUnits/Quantity.cs:15]: did this namespace change? If so why is this package not version 5?

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@pinzart90 I'm a bit concerned, maybe without reason about the namespace change? If others have referenced these types to use the unit types in their nodes won't they need to recompile their nodes on the new version?

Is ForgeUnits not semantically versioned?

**Comment by pinzart90** [comment]
> ~@pinzart90 I'm a bit concerned, maybe without reason about the namespace change? If others have referenced these types to use the unit types in their nodes won't they need to recompile their nodes on the new version?~
> 
> Oh I guess they will really only reference the Dynamo wrapper types, so it should be fine. Though my other question stands about ForgeUnits versioning strategy.
> 
> Is ForgeUnits not semantically versioned?

ForgeUnits is semantically versioned.
The move from 4.0.3 to 4.1.1 should not impact Dynamo API users that much, since (as you said) they only consume the Dynamo wrappers. 
About the namespace change: I had to change the namespace because the ForgeUnits package has 2 assemblies the old CLR dll and the new swig dll (I guess the FOrgeUnits devs want to support both for a while). If both dlls are referenced at compile time(which happens out of the box when referencing a nuget package) then we will get a namespace collision with the already defined alias in the dynamo code. There are other ways to fix it (I think), maybe to explicitely skip referencing the old CLR dll....but I chose to rename it

**Comment by mjkkirschner** [comment]
> ....

good info, thanks for sharing it.

File: src/Libraries/DynamoUnits/Quantity.cs (hunk 1 above)
Comment (pinzart90) [reply src/Libraries/DynamoUnits/Quantity.cs:15]: answered in the general chat

File: test/core/units/malformed_unit_dropdowns.dyn [hunk 2]
```diff
@@ -32,7 +32,7 @@
     },
     {
       "ConcreteType": "UnitsUI.Units, UnitsUI",
-      "SelectedIndex": 218,
```
Comment (pinzart90) [inline test/core/units/malformed_unit_dropdowns.dyn:35]: Looks like updating to the new forgeUnits will break some existing units nodes.
Looks like we are serializing the index in dropdown nodes which will break if the underlying list of items changes (which happened for this upgrade)
Do you guys think this is serious enough to try to migrate in some way ?
FYI @mjkkirschner @sm6srw

Comment (pinzart90) [reply test/core/units/malformed_unit_dropdowns.dyn:35]: Maybe this only happens when the dropDown's SelectedString property does not find a match in the existing itemList.
So an edge case...

Comment (pinzart90) [reply test/core/units/malformed_unit_dropdowns.dyn:35]: Ugh...it is weird. Looks like DropDown nodes serialize both the SelectedString and the SelectedIndex. No sure about the order of deserialization. When one is deserialized it tries to override the other....
https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/CoreNodeModels/DropDown.cs#L91-L147

Comment (pinzart90) [reply test/core/units/malformed_unit_dropdowns.dyn:35]: @QilongTang Looks like you implemented the serialization of DropDown SelectedString and SelectedIndex.
https://github.com/DynamoDS/Dynamo/pull/9367
Did you expect that the SelectedIndex might become out of sync with SelectedString ?

Comment (QilongTang) [reply test/core/units/malformed_unit_dropdowns.dyn:35]: SelectedString is deserialized after SelectedIndex, so when index is out of sync when item sources updated, changed, we fall back to SelectedString later as a second chance. Let me know if that is clear

Comment (pinzart90) [reply test/core/units/malformed_unit_dropdowns.dyn:35]: Ok..so this is only an issue when the SelectedString is no longer found. That means it was bad from the start .... or it is missing from the underlying item list, which would be a breaking change (so it might acceptable for it not to work properly)

**Review by sm6srw — APPROVED** [review]
